pysph/base/opencl.py

```diff
@@ -35,6 +42,31 @@ def set_queue(q):
     _queue = q
 
 
+def profile(name, event):
+    global _profile_info
+    event.wait()
+    time = (event.profile.end - event.profile.start) * 1e-9
+    if name in _profile_info:
+        _profile_info[name] += time
+    else:
```

Comment:
No need for the if block, since the defaultdict will always give you a default 0 float, you can just have `_profile_info[name] += time`